core/src/services/icloud/core.rs

```diff
@@ -110,9 +113,18 @@ impl Debug for IcloudSigner {
 }
 
 impl IcloudSigner {
-    /// Get the session data from signer.
-    pub fn session_data(&self) -> &SessionData {
-        &self.data
+    /// Get the drivews_url from signer session data.
+    /// Async await init finish.
+    pub async fn drivews_url(&mut self) -> Result<String> {
+        self.init().await?;
```

Comment:
Please check `self.initiated` first. Or we can move the check into `init()` directly.